client/client.go

```diff
@@ -442,7 +442,7 @@ func configureAwsClient(ctx context.Context, logger hclog.Logger, awsConfig *Con
 	// Test out retrieving credentials
 	if _, err := awsCfg.Credentials.Retrieve(ctx); err != nil {
 		logger.Error("error retrieving credentials", "err", err)
-		return awsCfg, fmt.Errorf(awsFailedToConfigureErrMsg, account.AccountName, err, checkEnvVariables())
+		return awsCfg, diag.FromError(fmt.Errorf(awsFailedToConfigureErrMsg, account.AccountName, err, checkEnvVariables()), diag.USER)
```

Comment:
This fixes a similar issue to https://github.com/cloudquery/cq-provider-aws/issues/869.
Happened to me on my default AWS profile which is my personal account. The default account failed to assume the role, so it reached this code.

Reply:
again, this is called by `client.Configure` which calls classifyError on the result. As long as the error is wrapped with %w the classification should be done in `classifyError`.

Reply:
Would it make sense to separate auth errors during the setup phase from auth errors during the actual resources resolving phase?

If we failed outright at the beginning with an auth error can we assume it's a user error?